add documentation (#2)
lapce-community/lapce-volt-template: README.md

# lapce-volt

This repository serves as a starting point for writing [Lapce](https://lapce.dev/) plugins in the [Rust Programming Language](https://www.rust-lang.org).

> If you haven't used Rust before, consider familiarizing yourself with the documentation available on .

Reviewer: ```suggestion
If you haven't used Rust before, consider familiarizing yourself with the [documentation](https://doc.rust-lang.org/book/) .
```

## Getting started

- If not yet done, [install Rust](https://www.rust-lang.org/learn/get-started)

> - This template uses `cargo-make` as a build system. Install it, too.

Reviewer: ```suggestion
- This template uses [cargo-make](https://github.com/sagiegurari/cargo-make) as a build system. Install it using `cargo install --force cargo-make`.
```

> - Generate your own repository based on this one by following the instructions [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template).

Reviewer: ```suggestion
- Ensure the wasi target is installed, using: `rustup target add wasm32-wasi`
- Generate your own repository based on this one by following the instructions [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template).
```

Project is unable to compile without this target added.

- [Clone it](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) locally

> - Compile your addon by executing `cargo make dev` in the cloned directory. This will create a `bin/lapce-plugin-template.wasm` file.

Reviewer: ```suggestion
- Ensure the wasi target is installed, using: `rustup target add wasm32-wasi`
- Compile your addon by executing `cargo make dev` in the cloned directory. This will create a `bin/lapce-plugin-template.wasm` file.
```

This target must be installed to compile.

- Edit the extension's metadata in the `volt.toml` file.
  - Check the `activation` block. Here, you can configure when Lapce will load your plugin. The two corresponding settings, `language` and `workspace-contains` accept an array of RegExps.
    - Although this is **not recommended because of performance concerns**, to always load your extension, you can use `*`
    - Edit the `wasm` property to reflect the (relative) path of the built plugin.

> - Lapce loads extensions that are placed inside its plugins folder.

Reviewer: ```suggestion
- Edit the `Cargo.toml` name property to match the `volt.toml` name property.
- Lapce loads extensions that are placed inside its plugins folder.
```

>   - If your filesystem supports symbolic links, symlink the plugin directory to `~/.local/share/lapce-stable/`

Reviewer: ```suggestion
  - If your filesystem supports symbolic links, symlink the plugin directory to `~/.local/share/lapce-stable/`, or `/Users/[USERNAME]/Library/Application Support/dev.lapce.Lapce-Stable/plugins` for Mac.
```
